tests/check/check.go

```diff
@@ -1239,6 +1258,84 @@ condition xcond(x: string) {
 							require.Equal(t, assertion.ErrorCode, int(e.Code()), detailedInfo)
 						}
 					})
+					t.Run(fmt.Sprintf("assertion_list_objects_%s", assertion.Name), func(t *testing.T) {
+						objectType, _ := tuple.SplitObject(assertion.Tuple.GetObject())
+						resp, err := client.ListObjects(ctx, &openfgav1.ListObjectsRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: modelID,
+							Type:                 objectType,
+							User:                 assertion.Tuple.GetUser(),
+							Relation:             assertion.Tuple.GetRelation(),
+							ContextualTuples: &openfgav1.ContextualTupleKeys{
+								// TODO
+								TupleKeys: []*openfgav1.TupleKey{},
+							},
+							Context: assertion.Context,
+						})
+
+						if assertion.ListObjectsErrorCode != 0 {
+							require.Error(t, err)
+							e, ok := status.FromError(err)
+							require.True(t, ok)
+							require.Equal(t, assertion.ListObjectsErrorCode, int(e.Code()))
+							return
+						}
+						if assertion.ErrorCode == 0 {
+							require.NoError(t, err)
+							if assertion.Expectation {
+								// >=, not >, because some users have access to more than one object
+								require.GreaterOrEqual(t, len(resp.GetObjects()), 1, "at least one object should be returned")
```

Comment:
why do we need this assertion?  Will the next line be sufficient?

Reply:
To be fair, the two assertions are testing for completely different conditions.

The first condition tests that _some_ result was returned, and provides a message that indicates this specific expectation. The second condition tests the the _correct_ result was returned, and provides a message that indicates this specific expectation.

If you leave out the first condition, the a failure of the second condition does not tell you that no result was returned, only that the _wrong_ result was returned. This would make it more difficult to isolate the problem.

Reply:
Exactly :)

Reply:
I am good either way.